main/src/main/java/cgeo/geocaching/log/LogTrackableActivity.java

```diff
@@ -258,35 +259,10 @@ public void onConfigurationChanged(@NonNull final Configuration newConfig) {
         init();
     }
 
-    @Override
-    public void onCreateContextMenu(final ContextMenu menu, final View view, final ContextMenu.ContextMenuInfo info) {
-        super.onCreateContextMenu(menu, view, info);
-        final int viewId = view.getId();
-
-        if (viewId == R.id.type) {
-            for (final LogTypeTrackable typeOne : possibleLogTypesTrackable) {
-                menu.add(viewId, typeOne.id, 0, typeOne.getLabel());
-            }
-        }
-    }
-
-    @Override
-    public boolean onContextItemSelected(final MenuItem item) {
-        final int group = item.getGroupId();
-        final int id = item.getItemId();
-
-        if (group == R.id.type) {
-            setType(LogTypeTrackable.getById(id));
-
-            return true;
-        }
-
-        return false;
-    }
-
     private void init() {
-        registerForContextMenu(binding.type);
-        binding.type.setOnClickListener(this::openContextMenu);
+        logType.setTextView(binding.type).setDisplayMapper(LogTypeTrackable::getLabel);
+        logType.setValues(possibleLogTypesTrackable);
+        logType.setChangeListener(this::setType);
 
         setType(typeSelected);
```

Comment:
Method "settype" should now be deletable. Instead the Textspinner's "set" value should be used. 

A good check in code is that the TextView "binding.type" should not be used anywhere in code except when assigned to the TextSpinner using it's "setTextView" method.

Reply:
Thanks for cross-checking. Method `setType` does a couple of things related to the selected value, so I guess it needs to stay. But I replaced a call to `binding.type.setText()` (now `logType.setTextView(binding.type)` is the only place left where `binding.type` is being used) and fixed a missing update to the spinner's content.